packages/mdctl-axon-tools/lib/StudyManifestTools.js

```diff
@@ -151,14 +179,29 @@ class StudyManifestTools {
     return { manifest, removedEntities }
   }
 
-  async buildManifestAndDependencies() {
+  async buildManifestAndDependencies(manifestObject) {
+    let ignoreKeys = [],
+        mappingScript,
+        study,
+        manifestJSON
+
     const { org, orgReferenceProps } = await this.getOrgAndReferences(),
-          study = await this.getFirstStudy(org),
-          allEntities = [study, ...await this.getStudyManifestEntities(org, study, orgReferenceProps)],
-          { manifest, removedEntities } = this.validateAndCreateManifest(allEntities, orgReferenceProps),
-          mappingScript = await getMappingScript(org),
           ingestTransform = fs.readFileSync(`${__dirname}/../packageScripts/ingestTransform.js`)
 
+    if (manifestObject) {
+      manifestJSON = this.validateManifest(manifestObject)
+
+      ignoreKeys = Object.keys(manifestJSON).filter(key => typeof manifestJSON[key] === 'object')
```

Comment:
I think this one can be improved slightly, overall it should work but we can make more secure

1. Notice that arrays are also considered objects so it will pick up the "objects" from the manifest if passed through, this should not happen but it may be a good idea to account for incorrect usage
2. I'd add part of this to `validateManifest` and may be rename it to `validateAndCleanManifest`

For example, inside validateAndCleanManifest:
```
return Object.keys(manifestJSON)
   .filter(key => availableObjectNames.includes(key))
   .reduce((acc,curr) => ({ ...acc, [curr]: manifestJSON[curr] })
```
The variable availableKeys can be another property returned by getOrgAndReferences , we are already getting all objects inside getOrgObjectInfo, to calculate availableObjectNames you just need to
```
//from this 
allObj = await objects.find()
            .limit(false)
            .toArray()

availableObjectNames = allObj.map(obj => obj.name)
```

After these changes creating the ignoreKeys would be matter of just doing:
```
ignoreKeys = Object.keys(manifestJSON)
```

Reply:
Finally if you decide to go with the approach above, the calculation of ignoreKeys can be simplified to:
```
ignoreKeys = Object.keys(manifestJSON)
        .map(key => this.mapObjectNameToPlural(key))
```